sync master

ca333 opened the pull request:
remove maven and use official repo

Commit:
Merge pull request #24 from KomodoPlatform/master

remove maven and use official repo

Commit:
adds missing version to example pubspec.yaml

Commit:
adds updated project build configs

Commit:
version bump 0.0.11

Commit:
fixes formatting of dart source files

chickahoona committed:
Update build.gradle

nsivraj committed:
Added the crypto_sign_ed25519_pk_to_curve25519 api for both ios and android.

nsivraj committed:
Removed a parenthesis causing a compile error.

nsivraj committed:
Fixed compile error that Joshua found on Android.

rohansingh committed:
Update podspec to specify Swift versions

ArtemGr commented on example/ios/Flutter/flutter_export_environment.sh:
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+# This is a generated file; do not edit or check into version control.
+export "FLUTTER_ROOT=/Users/kozw/flutter"
```
Note how some of the generated Flutter files are linked to the specifics of the developer work station, like paths.
I usually add such files into the `.gitignore`.

Commit:
initial ffi commit

Commit:
adds PasswordHash and RandomBytes

Commit:
fixes pwhash bindings

Commit:
adds PasswordHash hashStorageModerate and Sensitive functions

Commit:
refactors example app

Commit:
adds animation to result codeblock display

Commit:
adds android libsodium libs

Commit:
updates async hash sample

Commit:
fixes bindings for functions using size_t

Commit:
removes unneeded directive from android build.gradle and removes armeabi lib

Commit:
updates sample result animation in demo app

Commit:
adds generic hash

Commit:
re-adds missing demo app screenshot

Commit:
adds core and highlevel api samples

Commit:
moves samples from toc to separate samples.dart

Commit:
adds support for empty sample descriptions

Commit:
adds support for crypto_generichash streaming API

Commit:
Merge pull request #29 from chickahoona/upgrade-to-compilesdk-28

Upgraded compileSdkVersion to 28

Commit:
Merge pull request #30 from nsivraj/master

Added the crypto_sign_ed25519_pk_to_curve25519 api for both ios and android

Commit:
Merge pull request #34 from tidbyt/rohan/fix-swift

Update podspec to specify Swift versions

Commit:
updates readme

Commit:
fixes readme screenshot image

Commit:
adds .dart_tool to .gitignore

Commit:
adds primitives and implementation names to high-level API

Commit:
implements crypto_box authenticated encryption and sealed boxes

Commit:
refactors high-level API with consistent naming convention

Commit:
replaces local var with final

Commit:
fixes GenericHash multipart state management

Commit:
implements crypto_sign

Commit:
adds crypto_sign secret key extraction methods

Commit:
adds randomSeed helper functions

Commit:
moves samples to Samples class

Commit:
implements crypto_shorthash

Commit:
updates readme

Commit:
minor version sample fix

Commit:
replaces requireSuccess int extension with mustSucceed

Commit:
implements crypto_secretbox

Commit:
updates readme

Commit:
implements crypto_auth

Commit:
replaces var with final in samples

Commit:
removes primitive names from high-level API

Commit:
implements crypto_kdf

Commit:
implements crypto_kx

Commit:
implements crypto_scalarmult

Commit:
implements crypto_aead chacha20-poly1305 constructs
refactors bindings classes

Commit:
implements Ed25519 to Curve25519 key conversion

Commit:
implements crypto_onetimeauth

Commit:
implements crypto_hash

Commit:
replaces all " with '

Commit:
adds support for loading libsodium on osx
adds initial tests

Commit:
adds crypto_pwhash tests

Commit:
implements crypto_secretstream
removes Map from core API

Commit:
implements crypto_stream

Commit:
add FFI migration topic to readme

Commit:
simplifies variable names in samples

Commit:
adds a topic on background threads to readme

Commit:
updates version sample with sodium minimal details

Commit:
set Sodium.sodiumLibraryMinimal type to bool

Commit:
fixes opslimit and memlimit range check for Argon2i algorithm

Commit:
removes unused plugin files

Commit:
removes local config files

Commit:
version bump 0.1.0

Commit:
fixes formatting of dart files

Commit:
removes obsolete author field from pubspec.yaml

Commit:
Merge branch 'master' into ffi

Commit:
Merge pull request #37 from firstfloorsoftware/ffi

flutter_sodium FFI rewrite

Commit:
fixes incorrect parameter usage in key exchange sample

Commit:
adds dummy calls to prevent tree shaking in release

Commit:
version bump 0.1.1

Commit:
adds example markdown

Commit:
fixes multi-platform support warnings

Commit:
version bump 0.1.2

Commit:
fixes pubspec min flutter version

Commit:
reverts invalid pubspec.yaml platform updates

Commit:
version bump 0.1.3

Commit:
removes sodium prefix from version and init function names

Commit:
implements sodium_bin2hex and hex2bin helpers

Commit:
replaces all hex.encode with Sodium.bin2hex in samples

Commit:
implements sodium_bin2base64 and base642bin helpers

Commit:
fixes bin2hex support for ignore parameter having value null

Commit:
updates gitignore and build files

Commit:
Merge pull request #40 from firstfloorsoftware/sodium-helpers

Sodium helpers

Commit:
fixes #41 Error with hash function in Android 10/API 29

Commit:
version bump 0.1.4

Commit:
fixes #39 and #42

Commit:
version bump 0.1.5

Commit:
fixes compile error on Android where FlutterSodiumPlugin symbol could not be located

Commit:
version bump 0.1.6

Commit:
fixes android build warning 'FlutterSodiumPlugin.java uses or overrides a deprecated API'

Commit:
upgrades url_launcher dependency in example to fix deprecated android build warning

Commit:
removes obsolete convert dependency from flutter_sodium

Commit:
replaces convert hex encode/decode with sodium hex alternatives in test project

Commit:
fixes #44 removes file now in gitignore

Commit:
fixes #44 removes file now in gitignore

Commit:
adds Sodium.init documentation to readme

Commit:
minor readme update

Commit:
fixes #45 extends library documentation

Commit:
version bump 0.1.7

Commit:
first attempt at fixing #46 by replacing String with null terminated Uint8List in pwhash_str functions

Commit:
adds null terminator to str input when missing to Sodium.cryptoPwhashStrVerify and cryptoPwhashStrNeedsRehash

Commit:
version bump 0.1.8

Commit:
Merge pull request #47 from firstfloorsoftware/pwhash_str_null_terminated

Pwhash str null terminated

Commit:
fixes #48 READ_PHONE_STATE permission implicitly added on android

Commit:
version bump 0.1.9

Commit:
fixes build warning of example app

Commit:
adds support for sodium_runtime_*

Commit:
adds sodium_memcmp

Commit:
replaces ListEquality with new Sodium.memcmp in samples

Commit:
updates UInt8List toPointer extension to allocate an arbitrary size

Commit:
adds support for sodium_pad and sodium_unpad with samples

Commit:
version bump 0.1.10

Commit:
Merge pull request #50 from firstfloorsoftware/new-features

New features

Commit:
adds support for loading libsodium on linux

- verified with libsodium 1.0.18-stable on ubuntu 20.10

Commit:
Merge pull request #52 from efabens/add-linux-support

adds support for loading libsodium on linux

Commit:
migrates pubspec.yaml

Commit:
removes obsolete null asserts on non-nullable values

Commit:
fixes nullable extensions

Commit:
uses nullable parameters wherever needed
replaces @required attribute with required keyword
replaces utf8.encode with utf8.encoder.convert to avoid Uint8List casts

Commit:
implements null safety in example app

Commit:
version bump 0.2.0-nullsafety.0

Xoeseko committed:
Added suppport for loading libsodium on Windows

- verified with libsodium 1.0.18-stable on Windows 10 Pro version 1909 (OS build18363.1256)

Commit:
Merge pull request #55 from Xoeseko/master

Added suppport for loading libsodium on Windows

pheki committed:
Adds bindings to crypto_core_hchacha20 and crypto_core_hsalsa20

Commit:
Merge pull request #57 from pheki/core-bindings

Adds bindings to crypto_core_hchacha20 and crypto_core_hsalsa20

Commit:
adds crypto_core_hchacha and hsalsa to core API

Commit:
version bump 0.1.11

Commit:
Merge pull request #58 from firstfloorsoftware/crypto_core

Crypto core

Commit:
upgrades ffi dependency

Commit:
Merge branch 'master' into null-safety

Commit:
version bump 0.2.0-nullsafety.1

Commit:
migrates to ffi version 1.0.0

Commit:
migrates example app url_launcher to 6.0.2

Commit:
replaces obsolete RaisedButton with ElevatedButton in sample app

Commit:
version bump 0.2.0

Commit:
fixes dart beta sdk dependency

Commit:
Merge pull request #60 from firstfloorsoftware/null-safety

Stable null safety

Commit:
updates readme

proninyaroslav committed:
[Linux] Fix library load path

Commit:
Merge pull request #66 from proninyaroslav/linux

Using relative library path for Linux
